### PR TITLE
[WIP] Adding Search app.

### PR DESCRIPTION
**Pull request by kegor**
Replicated changes from: https://github.com/groupdocs-total/GroupDocs.Total-for-NET-MVC/pull/91.

### CODE REVIEW

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/client/apps/search-e2e/src/support/app.po.ts  1
- GroupDocs.Total.WebForms.Test/SearchTest.cs  1
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Service/SearchService.cs  13
- src/client/apps/search-e2e/src/plugins/index.js  1
- src/Search.aspx.cs  2
         

Clones added
============
- src/Products/Search/Util/Directory/DirectoryUtils.cs  2
- src/Products/Search/Controllers/SearchApiController.cs  5
- src/Search.aspx.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-WebForms/pullRequest?prid=5722228&bid=18720682)

File: src/Search.aspx.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using GroupDocs.Total.WebForms.Products.Common.Config;
+using System;
+using System.Web;
+
+namespace GroupDocs.Total.WebForms
+{
+    public partial class Search : System.Web.UI.Page
+    {
+        protected void Page_Load(object sender, EventArgs e)
```
Comment (bobkovalex) [inline src/Search.aspx.cs:9]: Codacy found an issue: [Make 'Page_Load' a static method.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-WebForms/pullRequest?prid=5722229)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/client/apps/search-e2e/src/support/app.po.ts  1
- GroupDocs.Total.WebForms.Test/SearchTest.cs  1
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Service/SearchService.cs  13
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Search.aspx.cs  2
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/client/apps/search-e2e/src/plugins/index.js  1
         

Clones added
============
- src/Products/Search/Util/Directory/DirectoryUtils.cs  2
- src/Search.aspx.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-WebForms/pullRequest?prid=5722229&bid=18720683)

File: src/Search.aspx.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using GroupDocs.Total.WebForms.Products.Common.Config;
+using System;
+using System.Web;
+
+namespace GroupDocs.Total.WebForms
+{
+    public partial class Search : System.Web.UI.Page
```
Comment (bobkovalex) [inline src/Search.aspx.cs:7]: Codacy found an issue: ['partial' is gratuitous in this context.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-WebForms/pullRequest?prid=5722229)

**Pull request merged**